### PR TITLE
Determine namespaceability of resources from openapi schema

### PR DESCRIPTION
**Pull request by mortent**
This uses the openapi schema to determine whether a resource is namespace-scoped or cluster-scoped as described [here](https://github.com/kubernetes-sigs/kustomize/issues/2861#issuecomment-676833212). 
Since up until now the openapi schema that was included in kpt was not complete, but only included type definitions, I had to update the openapi schema. It is still based on kubernetes version 1.17.
This only makes the logic available in kyaml, it does not update kustomize to use it.

Ref: #2861

### CODE REVIEW

File: kyaml/openapi/openapi.go [hunk 1]
```diff
@@ -368,27 +391,68 @@ func initSchema() {
...
+		gvk, found := pathInfo.Get.VendorExtensible.Extensions[kubernetesGVKExtensionKey]
+		if !found {
+			continue
+		}
+		typeMeta, found := toTypeMeta(gvk)
+		if !found {
+			continue
+		}
+
+		if strings.Contains(path, "namespaces/{namespace}") {
```
Comment (apelisse) [inline kyaml/openapi/openapi.go:446]: Yeah, you have two options:
1. Look for that specific pattern in the path, or
2. A path parameter like this: 
```
"in": "path",
"name": "namespace",
"required": true,
"type": "string",
```
3. Or both.

**Review by apelisse — COMMENTED** [review]
I think that's good. I don't know what is the best way to detect it from there looking for parameter vs string pattern. You could build a test that walks through all of them and compares?

File: kyaml/openapi/openapi.go (hunk 1 above)
Comment (mortent) [reply kyaml/openapi/openapi.go:446]: Yeah, I think both approaches should work equally well. The only tricky part is that even namespaced resources will have some paths that doesn't include the namespace path parameter (like `/apis/apps/v1/deployments`). But the code does account for this.

Comment (apelisse) [reply kyaml/openapi/openapi.go:446]: Right, that's fine! If you feel confident that this is good enough, go for it!
/lgtm

**Pull request merged**